packages/starknet-snap/src/rpcs/get-deployment-data.ts

```diff
@@ -0,0 +1,88 @@
+import type { Infer } from 'superstruct';
+import { object, string, assign, array } from 'superstruct';
+
+import {
+  AddressStruct,
+  BaseRequestStruct,
+  AccountRpcController,
+  CairoVersionStruct,
+} from '../utils';
+import { ACCOUNT_CLASS_HASH, CAIRO_VERSION } from '../utils/constants';
+import { AccountAlreadyDeployedError } from '../utils/exceptions';
+import {
+  getDeployAccountCallData,
+  isAccountDeployed,
+} from '../utils/starknetUtils';
+
+export const GetDeploymentDataRequestStruct = assign(
+  object({
+    address: AddressStruct,
+  }),
+  BaseRequestStruct,
+);
+
+export const GetDeploymentDataResponseStruct = object({
+  address: AddressStruct,
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  class_hash: string(),
+  salt: string(),
+  calldata: array(string()),
+  version: CairoVersionStruct,
+});
+
+export type GetDeploymentDataParams = Infer<
+  typeof GetDeploymentDataRequestStruct
+>;
+
+export type GetDeploymentDataResponse = Infer<
+  typeof GetDeploymentDataResponseStruct
+>;
+
+/**
+ * The RPC handler to get the deployment data.
+ */
```

Comment:
```suggestion
/**
 * The RPC handler to get the deployment data.
 * Handles force deploy (Cairo 0 with balance) and force upgrade (Cairo 0 without balance),
 * issuing alert in relevant scenarios.
 */
```

Reply:
if we add such comment, we have to add it everywhere